crates/data_components/src/kafka.rs

```diff
@@ -132,13 +134,95 @@ impl std::fmt::Debug for KafkaConfig {
                 "ssl_endpoint_identification_algorithm",
                 &self.ssl_endpoint_identification_algorithm,
             )
+            .field(
+                "metrics_store",
+                &self.metrics_store.as_ref().map(|_| "Some(KafkaMetrics)"),
+            )
             .finish()
     }
 }
 
+#[derive(Debug, Default)]
+pub struct KafkaMetrics {
+    /// Total consumer lag across all partitions
+    pub records_lag: AtomicU64,
+    /// Total number of messages consumed
+    pub records_consumed: AtomicU64,
+    /// Total bytes consumed
+    pub bytes_consumed: AtomicU64,
+}
+
+struct KafkaConsumerContext {
+    metrics: Arc<KafkaMetrics>,
+}
+
+impl KafkaConsumerContext {
+    fn new(metrics: Arc<KafkaMetrics>) -> Self {
+        Self { metrics }
+    }
+}
+
+impl KafkaMetrics {
+    #[must_use]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn update_records_lag(&self, lag: u64) {
+        self.records_lag.store(lag, Ordering::Relaxed);
+    }
+
+    pub fn update_records_consumed(&self, count: u64) {
+        self.records_consumed.store(count, Ordering::Relaxed);
+    }
+
+    pub fn update_bytes_consumed(&self, bytes: u64) {
+        self.bytes_consumed.store(bytes, Ordering::Relaxed);
+    }
+}
+
+impl rdkafka::ClientContext for KafkaConsumerContext {
+    #[allow(clippy::cast_sign_loss)]
+    fn stats(&self, statistics: rdkafka::Statistics) {
+        // Calculate total consumer lag from all topic partitions
+        let mut total_lag = 0u64;
+        let mut has_valid_partitions = false;
+
+        for topic in statistics.topics.values() {
```

Comment:
Does this calculate for the entire Kafka cluster or just the topics this consumer is subscribed to?